gio/src/task.rs

```diff
@@ -99,44 +273,23 @@ impl Task {
             let task_func: Box_<Q> = Box::from_raw(user_data as *mut _);
             task_func(
                 task.as_ref(),
-                source_object.as_ref().as_ref(),
+                source_object.as_ref().as_ref().map(|s| s.unsafe_cast_ref()),
```

Comment:
double check this, because I am not sure about the use of unsafe cast

Reply:
That's correct